kubernetes-client-openapi-informer/src/main/java/io/micronaut/kubernetes/client/openapi/informer/InformerApiCallFactory.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client.openapi.informer;
+
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.type.Argument;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.inject.BeanDefinition;
+import io.micronaut.inject.ExecutableMethod;
+import io.micronaut.inject.qualifiers.Qualifiers;
+import io.micronaut.kubernetes.client.openapi.common.KubernetesObject;
+import io.micronaut.kubernetes.client.openapi.reactor.annotation.KubernetesClientApiReactor;
+import io.micronaut.kubernetes.client.openapi.watcher.annotation.KubernetesClientApiWatcher;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * The factory which creates {@link InformerApiCall} instances using mappings between
+ * kubernetes api types and {@link ExecutableMethod} instances which can be used for
+ * execution of {@code list} and {@code watch} api calls.
+ */
+@SuppressWarnings("rawtypes")
+@Singleton
+final class InformerApiCallFactory {
```

Comment:
Seems like this would be better written as implementing `ExecutableMethodProcessor<KubernetesClientApiReactor>` and implement `process` which is called on startup

Reply:
Modified, thanks